src/compas_viewer/_viewer.py

```diff
@@ -0,0 +1,41 @@
+import sys
+
+from PySide6.QtWidgets import QApplication
+
+from compas_viewer.components.renderer import Renderer
+from compas_viewer.config import Config
+from compas_viewer.configurations import ControllerConfig
+from compas_viewer.configurations import RendererConfig
+from compas_viewer.controller import Controller
+from compas_viewer.scene.scene import ViewerScene
+from compas_viewer.ui.ui import UI
+
+
+class Viewer:
+    _instance = None
+
+    def __new__(cls, *args, **kwargs):
+        if not cls._instance:
+            cls._instance = super().__new__(cls)
+            cls._is_initialised = False
+        return cls._instance
+
+    def __init__(self, *args, **kwargs):
+        if not self._is_initialised:
+            if not QApplication.instance():
+                self.app = QApplication(sys.argv)
+            else:
+                self.app = QApplication.instance()
```

Comment:
I have the feeling the second `if` is not needed. Because Viewer is already a singleton, the code inside first `if` will never be called twice.
```suggestion
        if not self._is_initialised:
            self.app = QApplication(sys.argv)
```